lib/ansible/modules/packaging/os/dnf.py

```diff
@@ -334,16 +334,29 @@ def is_lockfile_pid_valid(self):
         # https://github.com/ansible/ansible/issues/57189
         return True
 
-    def _sanitize_dnf_error_msg(self, spec, error):
+    def _sanitize_dnf_error_msg_install(self, spec, error):
         """
         For unhandled dnf.exceptions.Error scenarios, there are certain error
-        messages we want to filter. Do that here.
+        messages we want to filter in an install scenario. Do that here.
         """
         if to_text("no package matched") in to_text(error):
             return "No package {0} available.".format(spec)
 
         return error
 
+    def _sanitize_dnf_error_msg_remove(self, spec, error):
+        """
+        For unhandled dnf.exceptions.Error scenarios, there are certain error
+        messages we want to ignore in a removal scenario as known benign
+        failures. Do that here.
+        """
+        if 'no package matched' in to_native(error):
```

Comment:
Note: we end up calling `to_native(error)` in both the caller and inside the function.  We should change it so that the call is only done outside and we document that an error message is passed in via error or the call is only done inside of the function (and is done earlier top so that both the True and False return values are guaranteed to return a string).